specs/NavigationViewTemplateSettings-OpenPaneLength.md

```diff
@@ -0,0 +1,94 @@
+NavigationViewTemplateSettings.OpenPaneLength
+===
+
+# Background
+
+_This spec adds an `OpenPaneLength` property to the Xaml
+[NavigationViewTemplateSettings](https://docs.microsoft.com//windows/winui/api/microsoft.ui.xaml.controls.navigationviewtemplatesettings) class_
+
+The (existing) WinUI [NavigationView](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.NavigationView) control has
+a list of items you can view, and a content area that shows the selected item.
+The list is shown in a pane that can be open, closed or partially open (compact).
+The exact width of the open pane can be specified using the
+[NavigationView.OpenPaneLength](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.NavigationView.OpenPaneLength)
+property.
+
+Example of compact pane:
+
+![example of compact pane](images/nav-pane-closed.jpg)
+
+Example of open pane:
+
+![example of open pane](images/nav-pane-open.jpg)
+
+The new property in this spec is a read-only `NavigationViewTemplateSettings.OpenPaneLength`,
+to go along with the existing read/write `NavigationView.OpenPaneLength`.
+It returns the `min` between `NavigationView.OpenPaneLength` and the window width.
+This was necessary to fix a bug where the NavigationView pane contents were being cropped when
+`OpenPaneLength` surpasses the window's width.
+`NavigationViewTemplateSettings.OpenPaneLength` is set in `SplitView` and `ShadowCaster` template;
+it was previously a simple template binding to `NavigationView.OpenPaneLength` property.
+
+(Naming note: these APIs use "Length" rather than "Width" because
+in the original design it was envisioned that there woould be a vertical mode in the future,
+where this could represent a height too.)
+
+TemplateSettings properties are not commonly used by app developers, 
+however, in situations when they copy a WinUI control's template into their own project,
+there is a Xaml error when the TemplateSettings property is not public.
+See [issue #6682](https://github.com/microsoft/microsoft-ui-xaml/issues/6682)
+for more information on the error caused by non-public TemplateSettings.
+
+# API Pages
+
+## NavigationViewTemplateSettings.OpenPaneLength
+
+Gets the `min` between `OpenPaneLength` and the window's width.
+This is the calculated value of the width of the pane when opened and fully expanded. Defaults to 320.0.
```

Comment:
Should we document the units too? Or is it generally assumed for all these sort of properties that they're DIPs and we don't need to explicitly say that?

Reply:
Units are an important detail and should be specified.

Reply:
Added 'in view pixels' for this spec. Also making similar update in the ref topic.

Reply:
Yes, the ref topic should have the same detail.